server/schedule/checker/replica_checker.go

```diff
@@ -62,37 +64,49 @@ func (r *ReplicaChecker) GetType() string {
 }
 
 // Check verifies a region's replicas, creating an operator.Operator if need.
-func (r *ReplicaChecker) Check(region *core.RegionInfo) *operator.Operator {
+func (r *ReplicaChecker) Check(region *core.RegionInfo) (op *operator.Operator) {
 	checkerCounter.WithLabelValues("replica_checker", "check").Inc()
-	if op := r.checkDownPeer(region); op != nil {
-		checkerCounter.WithLabelValues("replica_checker", "new-operator").Inc()
-		op.SetPriorityLevel(core.HighPriority)
-		return op
+	abnormalCount := 0
+	if downCount, storeID := r.checkDownPeer(region); downCount > 0 {
+		op = r.fixPeer(region, storeID, downStatus)
+		abnormalCount = abnormalCount + downCount
 	}
-	if op := r.checkOfflinePeer(region); op != nil {
-		checkerCounter.WithLabelValues("replica_checker", "new-operator").Inc()
-		op.SetPriorityLevel(core.HighPriority)
-		return op
+	if offlineCount, storeID := r.checkOfflinePeer(region); offlineCount > 0 {
+		if op == nil {
+			op = r.fixPeer(region, storeID, offlineStatus)
+		}
+		abnormalCount = abnormalCount + offlineCount
 	}
-	if op := r.checkMakeUpReplica(region); op != nil {
+	if miss := r.checkMakeUpReplica(region); miss > 0 {
+		if op == nil {
+			op = r.fixMakeUpReplica(region)
+		}
+		abnormalCount = abnormalCount + miss
+	}
+	if op != nil {
 		checkerCounter.WithLabelValues("replica_checker", "new-operator").Inc()
 		op.SetPriorityLevel(core.HighPriority)
+		if abnormalCount > r.opts.GetMaxReplicas()/2 {
+			r.regionPriorityQueue.Push(abnormalCount, region.GetID())
```

Comment:
A new function can be added to PriorityQueue: if the queue is full, the element at the end of the queue (not top, which is equivalent to randomly selecting a lower priority) element is selected for comparison, and if the element to be inserted has a higher priority, it will be discarded, first delete the tail element.

Reply:
yes, it should insert if the tail has low priority